### DISCUSSION TITLE
3.10.0: Add QZSS time scale -- possibly a breaking change

### DISCUSSION THREAD

**Discussion by ChristopherRabotin** [nyx-space/hifitime#293] [Announcements]
## What's next?
This is probably the last release of version 3.x. There's a fair amount of small bugs here and there in 3.x, and these aren't trivial to fix. The architecture changes introduced by @gwbres and I for the development of 4.x should fix all of those in one go.

Also, version 4.0 will likely switch to the Mozilla Public License from Apache 2: if you have concerns about this, please head over to https://github.com/nyx-space/hifitime/discussions/274 to voice them. From the user point of view, there should be absolutely no change, even if you're selling code that uses an unmodified version of Hifitime.

## What's Changed
* support qzss time scale by @gwbres in https://github.com/nyx-space/hifitime/pull/228
* Update 4.0.0-dev to latest master by @ChristopherRabotin in https://github.com/nyx-space/hifitime/pull/243
* Integrate 3.8.6 into 4.0.0-dev by @ChristopherRabotin in https://github.com/nyx-space/hifitime/pull/264
* BREAKING CHANGE: 4.0.0-dev branch become 3.10.0 by @ChristopherRabotin in https://github.com/nyx-space/hifitime/pull/277
* Draft: replace std::time:SystemTime with web_time::SystemTime by @karpfediem in https://github.com/nyx-space/hifitime/pull/279
* Update reqwest requirement from 0.11 to 0.12 by @dependabot in https://github.com/nyx-space/hifitime/pull/286
* Fix panic caused by arithmetic overflow by @cardigan1008 in https://github.com/nyx-space/hifitime/pull/283
* Fix panic caused by array out of bounds by @cardigan1008 in https://github.com/nyx-space/hifitime/pull/284
* Fix panic caused by unwrap by @cardigan1008 in https://github.com/nyx-space/hifitime/pull/285

## New Contributors
* @karpfediem made their first contribution in https://github.com/nyx-space/hifitime/pull/279
* @cardigan1008 made their first contribution in https://github.com/nyx-space/hifitime/pull/283

**Full Changelog**: https://github.com/nyx-space/hifitime/compare/3.9.0...3.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/nyx-space/hifitime/releases/tag/3.10.0'>3.10.0: Add QZSS time scale -- possibly a breaking change</a>.</em>

**Comment by gwbres** [comment]
Hello Chris, this looks very promising :rocket:

Right now I'm working on PPP performances enhancement, I want to conclude this topic prior taking the leap.
I'm also unsure what the best strategy would be to take the leap, considering hifitime is core library to us, and impacts at every level

**Reply by gwbres** [reply]
Hello @ChristopherRabotin,

I cannot put the new Hifitime to the test using my RTK solver (the ultimate usecase) until  `nyx` gets updated. 
Would it be possible to propose a new nyx-dev branch ?
All I can say is I have updated all my crates manually, bypassed the RTK stuff, ran the RINEX CI and it did pass 🚀 🚀  :rocket:
Considering we have about 230 tests at the moment, that is quite something to say

**Reply by ChristopherRabotin** [reply]
Ask and you shall receive: https://github.com/nyx-space/nyx/pull/294 !

**Reply by gwbres** [reply]
Will hifitime V4 come with more friendly Serdes Input/output ?

**Comment by ChristopherRabotin** [comment]
Indeed! In fact, that change is already merged to master as of yesterday.

**Reply by gwbres** [reply]
Ok cool that is a good thing. 
For example I use JSON configuration files in my application and the previous behavior kind of forbid using `Durations` in those, due to the impossibility to understand the value
